ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/StartupAwaitDeploymentUnitProcessor.java

```diff
@@ -0,0 +1,66 @@
+package org.jboss.as.ejb3.deployment.processors;
+
+import org.jboss.as.ee.component.Attachments;
+import org.jboss.as.ee.component.ComponentConfiguration;
+import org.jboss.as.ee.component.ComponentConfigurator;
+import org.jboss.as.ee.component.ComponentDescription;
+import org.jboss.as.ee.component.EEModuleDescription;
+import org.jboss.as.ee.component.ViewConfiguration;
+import org.jboss.as.ee.component.interceptors.InterceptorOrder;
+import org.jboss.as.ejb3.component.EJBComponentDescription;
+import org.jboss.as.ejb3.component.EJBViewConfiguration;
+import org.jboss.as.ejb3.component.MethodIntf;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.invocation.ImmediateInterceptorFactory;
+import org.jboss.invocation.InterceptorFactory;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Adds  StartupAwaitInterceptor to exposed methods of EJB, forcing users to wait until all startup beans in the deployment are done with post-construct methods.
+ * @author Fedor Gavrilov
+ */
+// adding an abstraction for the whole deployment unit to depend on while blocking external client calls is a better solution probably
+// it requires a lot of rewriting in EJB code right now, hence this class to satisfy EJB 3.1 spec, section 4.8.1
+// feel free to remove this class as well as StartupAwaitInterceptor and StartupCountDownInterceptor when if easier way to satisfy spec will appear
+public class StartupAwaitDeploymentUnitProcessor implements DeploymentUnitProcessor {
+    private static final Set<MethodIntf> INTFS = EnumSet.of(MethodIntf.MESSAGE_ENDPOINT, MethodIntf.REMOTE, MethodIntf.SERVICE_ENDPOINT);
```

Comment:
This is not enought to satisfy the spec. Local invocation could be delivered through a servlet that injects the local interface for example.